### PR TITLE
fix: creating a Parse.File with base64 string fails for some encodings

### LINKED ISSUE #1510: Creating a Parse.File with base64 string fails
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/Parse-SDK-JS/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the latest versions of [Parse Server](https://github.com/parse-community/parse-server/releases) and the [Parse JS SDK](https://github.com/parse-community/Parse-SDK-JS/releases). 

### Issue Description
When Calling new Parse.File with the versions 3.4.1 upwards the process takes a few mins then actually throws an error saying 
"Cannot create a Parse.File without valid data URIs or base64 encoded data."

### Steps to reproduce
Calling Parse.file with the file blob will reproduce the issue

### Actual Outcome
an error gets thrown saying: "Cannot create a Parse.File without valid data URIs or base64 encoded data."

### Expected Outcome
getting a Parse File back

###  Environment
not a server issue but => 

Server
- Parse Server version: `n/a`
- Operating system: `n/a`
- Local or remote host (AWS, Azure, Google Cloud, Heroku, Digital Ocean, etc): `n/a`

Database
- System (MongoDB or Postgres): `n/a`
- Database version: `n/a`
- Local or remote host (MongoDB Atlas, mLab, AWS, Azure, Google Cloud, etc): `n/a`

Client
- Parse JS SDK version: `3.4.3`

### Logs
Screenshot 2022-07-14 at 15 29 26

### PR DESCRIPTION
**Pull request by justinlettau**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/Parse-SDK-JS/issues?q=is%3Aissue).

### Issue Description

Related issue: #1510

### Approach

Adding a failing unit test for reproducing base64 regex issue.

### TODOs before merging

- [x] Add tests

### CODE REVIEW

File: src/ParseFile.js [hunk 1]
```diff
@@ -43,7 +43,7 @@ export type FileSource =
     };
 
 const base64Regex = new RegExp(
-  '([0-9a-zA-Z+/]{4})*(([0-9a-zA-Z+/]{2}==)|([0-9a-zA-Z+/]{3}=))',
+  '([0-9a-zA-Z+/]{4})*(([0-9a-zA-Z+/]{2}==)|([0-9a-zA-Z+/]{3}=)|([0-9a-zA-Z+/]{4}))',
```
Comment (mtrezza) [inline src/ParseFile.js]:
```suggestion
  '([0-9a-zA-Z+/]{4})*(([0-9a-zA-Z+/][AQgw]==)|([0-9a-zA-Z+/]{2}[AEIMQUYcgkosw048]=)|([0-9a-zA-Z+/]{4}))',
```

**Review by mtrezza — CHANGES_REQUESTED** [review]

**Review by mtrezza — APPROVED** [review]
Looks good!

**Comment by mtrezza** [comment]
off-topic: @dblythy it's interesting that the CI passes here, compared to https://github.com/parse-community/Parse-SDK-JS/issues/1515

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [3.4.4-alpha.1](https://github.com/parse-community/Parse-SDK-JS/releases/tag/3.4.4-alpha.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [3.5.0-beta.1](https://github.com/parse-community/Parse-SDK-JS/releases/tag/3.5.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [3.5.0](https://github.com/parse-community/Parse-SDK-JS/releases/tag/3.5.0)

**Pull request merged**